### ISSUE TITLE
Cannot install on Razzberry Pi Zero W

### ISSUE THREAD

**Issue by Arkadije** [mKeRix/room-assistant#960]
**Describe the bug**
When trying to install RA on RPi Zero W following [This](https://www.room-assistant.io/guide/quickstart-pi-zero-w.html#installing-room-assistant)
Getting these warning messages

```
pi@raspberrypi:~ $ sudo npm i --global --unsafe-perm room-assistant
npm WARN deprecated har-validator@5.1.5: this library is no longer supported
npm WARN deprecated uuid@3.4.0: Please upgrade to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
npm WARN deprecated tar@2.2.2: This version of tar is no longer supported, and will not receive security updates. Please upgrade asap.

added 327 packages, and audited 328 packages in 36m

21 packages are looking for funding
  run `npm fund` for details

16 vulnerabilities (12 moderate, 4 high)

To address issues that do not require attention, run:
  npm audit fix

To address all issues (including breaking changes), run:
  npm audit fix --force

Run `npm audit` for details.

```

**Relevant configuration**
N/A not getting that far

**Expected behavior**
The command to complete without errors so that I can continue through the documentation

**Environment**
installation type: NodeJS
hardware: Raspberry Pi Zero W
OS: Raspberry Pi OS (Lite)

**Additional context - end of log file**
```
303 info run epoll@4.0.1 install node_modules/room-assistant/node_modules/epoll node-gyp rebuild
2304 info run epoll@4.0.1 install { code: 0, signal: null }
2305 timing build:run:install:node_modules/room-assistant/node_modules/epoll Completed in 49644ms
2306 info run i2c-bus@5.2.2 install node_modules/room-assistant/node_modules/i2c-bus node-gyp rebuild
2307 info run i2c-bus@5.2.2 install { code: 0, signal: null }
2308 timing build:run:install:node_modules/room-assistant/node_modules/i2c-bus Completed in 70741ms
2309 info run mdns@2.7.2 install node_modules/room-assistant/node_modules/mdns node-gyp rebuild
2310 verbose stack Error: write EIO
2310 verbose stack     at afterWriteDispatched (node:internal/stream_base_commons:164:15)
2310 verbose stack     at writeGeneric (node:internal/stream_base_commons:155:3)
2310 verbose stack     at WriteStream.Socket._writeGeneric (node:net:795:11)
2310 verbose stack     at WriteStream.Socket._write (node:net:807:8)
2310 verbose stack     at writeOrBuffer (node:internal/streams/writable:389:12)
2310 verbose stack     at _write (node:internal/streams/writable:330:10)
2310 verbose stack     at WriteStream.Writable.write (node:internal/streams/writable:334:10)
2310 verbose stack     at Gauge._doRedraw (/opt/nodejs/lib/node_modules/npm/node_modules/gauge/index.js:226:22)
2310 verbose stack     at Timeout._onTimeout (/opt/nodejs/lib/node_modules/npm/node_modules/gauge/index.js:15:19)
2310 verbose stack     at listOnTimeout (node:internal/timers:557:17)
2311 verbose cwd /home/pi
2312 verbose Linux 5.10.63+
2313 verbose argv "/opt/nodejs/bin/node" "/usr/local/bin/npm" "i" "--global" "--unsafe-perm" "room-assistant"
2314 verbose node v16.13.0
2315 verbose npm  v8.1.2
2316 error code EIO
2317 error syscall write
2318 error errno -5
2319 error write EIO
2320 verbose exit -5
```

Please advise

**Comment by bigstation17** [comment]
Same here.

**Comment by JeroenTuinstra** [comment]
Try what it says in the notice:

```
npm notice
npm notice New patch version of npm available! 8.1.0 -> 8.1.2
npm notice Changelog: https://github.com/npm/cli/releases/tag/v8.1.2
npm notice Run npm install -g npm@8.1.2 to update!
npm notice
```
`sudo npm install -g npm@8.1.2`

And then do `sudo npm i --global --unsafe-perm room-assistant` again.

Had similar errors when upgrading room assistant.

**Comment by Arkadije** [comment]
Ths that did it.

**Comment by bigstation17** [comment]
`pi@RoomAssistant:~ $ sudo npm i --global --unsafe-perm room-assistant
npm WARN deprecated har-validator@5.1.5: this library is no longer supported
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circums                         tances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
npm WARN deprecated tar@2.2.2: This version of tar is no longer supported, and will not receive security updates. Please upgrade                          asap.
npm WARN cleanup Failed to remove some directories [
npm WARN cleanup   [
npm WARN cleanup     '/opt/nodejs/lib/node_modules/room-assistant/node_modules/send',
npm WARN cleanup     [Error: Unknown system error -117: Unknown system error -117, unlink '/opt/nodejs/lib/node_modules/room-ass                         istant/node_modules/send/LICENSE'] {
npm WARN cleanup       errno: -117,
npm WARN cleanup       code: 'Unknown system error -117',
npm WARN cleanup       syscall: 'unlink',
npm WARN cleanup       path: '/opt/nodejs/lib/node_modules/room-assistant/node_modules/send/LICENSE'
npm WARN cleanup     }
npm WARN cleanup   ],
npm WARN cleanup   [
npm WARN cleanup     '/opt/nodejs/lib/node_modules/room-assistant',
npm WARN cleanup     [Error: Unknown system error -117: Unknown system error -117, unlink '/opt/nodejs/lib/node_modules/room-ass                         istant/dist/app.config.js'] {
npm WARN cleanup       errno: -117,
npm WARN cleanup       code: 'Unknown system error -117',
npm WARN cleanup       syscall: 'unlink',
npm WARN cleanup       path: '/opt/nodejs/lib/node_modules/room-assistant/dist/app.config.js'
npm WARN cleanup     }
npm WARN cleanup   ]
npm WARN cleanup ]
npm notice
npm notice New patch version of npm available! 8.1.0 -> 8.1.3
npm notice Changelog: https://github.com/npm/cli/releases/tag/v8.1.3
npm notice Run npm install -g npm@8.1.3 to update!
npm notice
npm ERR! code 1
npm ERR! path /opt/nodejs/lib/node_modules/room-assistant
npm ERR! command failed
npm ERR! command sh -c github-sponsors
npm ERR! node:internal/modules/cjs/loader:936
npm ERR!   throw err;
npm ERR!   ^
npm ERR!
npm ERR! Error: Cannot find module './dist'
npm ERR! Require stack:
npm ERR! - /opt/nodejs/lib/node_modules/room-assistant/node_modules/yaml/index.js
npm ERR! - /opt/nodejs/lib/node_modules/room-assistant/node_modules/github-sponsors/lib/github-sponsors.cjs.js
npm ERR! - /opt/nodejs/lib/node_modules/room-assistant/node_modules/github-sponsors/bin/github-sponsors.js
npm ERR!     at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)
npm ERR!     at Function.Module._load (node:internal/modules/cjs/loader:778:27)
npm ERR!     at Module.require (node:internal/modules/cjs/loader:1005:19)
npm ERR!     at require (node:internal/modules/cjs/helpers:102:18)
npm ERR!     at Object.<anonymous> (/opt/nodejs/lib/node_modules/room-assistant/node_modules/yaml/index.js:1:18)
npm ERR!     at Module._compile (node:internal/modules/cjs/loader:1101:14)
npm ERR!     at Object.Module._extensions..js (node:internal/modules/cjs/loader:1153:10)
npm ERR!     at Module.load (node:internal/modules/cjs/loader:981:32)
npm ERR!     at Function.Module._load (node:internal/modules/cjs/loader:822:12)
npm ERR!     at Module.require (node:internal/modules/cjs/loader:1005:19) {
npm ERR!   code: 'MODULE_NOT_FOUND',
npm ERR!   requireStack: [
npm ERR!     '/opt/nodejs/lib/node_modules/room-assistant/node_modules/yaml/index.js',
npm ERR!     '/opt/nodejs/lib/node_modules/room-assistant/node_modules/github-sponsors/lib/github-sponsors.cjs.js',
npm ERR!     '/opt/nodejs/lib/node_modules/room-assistant/node_modules/github-sponsors/bin/github-sponsors.js'
npm ERR!   ]
npm ERR! }

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2021-11-08T11_27_40_851Z-debug.log
pi@RoomAssistant:~ $ npm install -g npm@8.1.3
/opt/nodejs/lib/node_modules/npm/lib/utils/unsupported.js:28
    log.warn('npm', 'npm does not suppconst semver = require('semver')
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

SyntaxError: missing ) after argument list
    at Object.compileFunction (node:vm:352:18)
    at wrapSafe (node:internal/modules/cjs/loader:1031:15)
    at Module._compile (node:internal/modules/cjs/loader:1065:27)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1153:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at module.exports (/opt/nodejs/lib/node_modules/npm/lib/cli.js:10:7)
    at Object.<anonymous> (/opt/nodejs/lib/node_modules/npm/bin/npm-cli.js:2:25)
`

This errors i get on fresh install of Raspbian and guide step by step :(

**Comment by JeroenTuinstra** [comment]
Is see you did `npm install -g npm@8.1.3` maybe try: `sudo npm install -g npm@8.1.3`
But it seems you have lot of other errors. You sure you did a fresh install according to the guide? There seems to be a lot missing. 
You also seem to have some other packages that are out-dated: 
```
har-validator@5.1.5
uuid@3.4.0
request@2.88.2
tar@2.2.2
```

Try first `sudo apt update && sudo apt full-upgrade`

This will update your Raspbian first. Reboot and update npm and then try `sudo npm i --global --unsafe-perm room-assistant` again.
